From stable-bounces@linux.kernel.org  Tue Jan 31 21:27:10 2006
Date: Tue, 31 Jan 2006 21:20:26 -0800
From: Linus Torvalds <torvalds@osdl.org>
To: stable@kernel.org
Cc: 
Subject: seclvl settime fix

Don't try to "validate" a non-existing timeval.

settime() with a NULL timeval is silly but legal.

Noticed by Dave Jones <davej@redhat.com>

Signed-off-by: Linus Torvalds <torvalds@osdl.org>
[chrisw: seclvl only]
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 security/seclvl.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.15.2/security/seclvl.c
===================================================================
--- linux-2.6.15.2.orig/security/seclvl.c
+++ linux-2.6.15.2/security/seclvl.c
@@ -369,7 +369,7 @@ static int seclvl_capable(struct task_st
 static int seclvl_settime(struct timespec *tv, struct timezone *tz)
 {
 	struct timespec now;
-	if (seclvl > 1) {
+	if (tv && seclvl > 1) {
 		now = current_kernel_time();
 		if (tv->tv_sec < now.tv_sec ||
 		    (tv->tv_sec == now.tv_sec && tv->tv_nsec < now.tv_nsec)) {
